domain/machine/state/machine_cloud_instance.go

```diff
@@ -114,7 +125,7 @@ WHERE  availability_zone.name = $availabilityZoneName.name
 			instanceData.CPUCores = hardwareCharacteristics.CpuCores
 			instanceData.CPUPower = hardwareCharacteristics.CpuPower
 			instanceData.VirtType = hardwareCharacteristics.VirtType
-			if hardwareCharacteristics.AvailabilityZone != nil {
+			if hardwareCharacteristics.AvailabilityZone != nil && *hardwareCharacteristics.AvailabilityZone != "" {
```

Comment:
**thought:** Seems that nil AZ have the same semantic thant empty AZ every where. Why not simply remove the pointer ?

Reply:
You are right, we could change that in a future PR, it's an old struct used in many places.

Reply:
The intent is to be able to differentiate between an empty AZ and the absence of one.  Although in reality, I'm not sure that ever manifested.